Inc/ST-LIB_HIGH/Protections/Boundary.hpp

```diff
@@ -57,57 +65,88 @@ template<class Type, ProtectionType Protector> struct Boundary;
 template<class Type>
 struct Boundary<Type, BELOW> : public BoundaryInterface{
 	static constexpr ProtectionType Protector = BELOW;
+	bool has_warning_level{false};
 	Type* src = nullptr;
 	Type boundary;
-	// we have to do this because we cannot take address of rvalue (ProtectionType::BELOW)
-	
-	
-	Boundary(Type boundary):boundary(boundary)
-	{
-		
-	}
+	Type warning_threshold;
+	constexpr Boundary(const Type warn, const Type bound)
```

Comment:
i would add bound before warn, thus if we decide to add a constructor without warn value sintax stays consistent

Reply:
there is already a constructor taking only one value, check line 79. If you only pass one value then it is a fault, otherwise first value will be the warning.

Reply:
putting warn before is just a logical rule in my opinion, warning should be reached before fault. As i knew it could be of issue I tried the static check but had to conform with a run time check with ErrorHandler